lib/modules/manager/npm/post-update/pnpm.ts

```diff
@@ -82,18 +100,7 @@ export async function generateLockFile(
 
     let args = '--lockfile-only';
 
-    // pnpm v9+ is doing recursive automatically when it detects workspaces.
-    //
-    // If it's not a workspaces project/monorepo, but single project with unrelated other npm project in source tree (for example, a git submodule),
-    // `--recursive` will install un-wanted project.
-    // we should avoid this.
-    if (
-      pnpmToolConstraint.constraint &&
-      !semver.intersects(pnpmToolConstraint.constraint, '>=9') &&
-      (await localPathExists(pnpmWorkspaceFilePath))
-    ) {
-      args += ' --recursive';
-    }
+    const isWorkspace = await isPnpmWorkspace(pnpmWorkspaceFilePath);
```

Comment:
i wonder if we already extracted this information and added to manager config. so we don't need to parse the file twice (perf)